lib/md059.mjs

```diff
@@ -0,0 +1,41 @@
+// @ts-check
+
+import { addErrorContext, stripAndDowncaseText } from "../helpers/helpers.cjs";
+import { validLink } from "../helpers/micromark-helpers.cjs";
+import { getReferenceLinkImageData, filterByTypesCached } from "./cache.mjs";
+
+const defaultBannedNames = [
+  "read more",
+  "learn more",
+  "more",
+  "here",
+  "click here",
+  "link"
+];
+
+/** @type {import("markdownlint").Rule} */
+export default {
+  "names": [ "MD059", "no-generic-link-name" ],
+  "description": "Links should have a descriptive name",
+  "tags": [ "links", "accessibility" ],
+  "parser": "micromark",
+  "function": function MD059(params, onError) {
+    const { definitions } = getReferenceLinkImageData();
+
+    // Allow consumer to define their own banned names list.
+
+    const bannedNames = params.config.banned_names_override || defaultBannedNames;
+    const labels = filterByTypesCached([ "label" ])
+      .filter((label) => label.parent?.type === "link");
+
+    for (const label of labels) {
+      const labelTexts = label.children.filter((child) => child.type === "labelText");
+      for (const labelText of labelTexts) {
+        const text = labelText.text;
+        if (validLink(label, labelText, definitions) && bannedNames.includes(stripAndDowncaseText(text))) {
+          addErrorContext(onError, labelText.startLine, text);
```

Comment:
We can provide specific context and range information, too, I think - see what MD039 does.

Reply:
TODO

Reply:
Updated in [02d1166](https://github.com/DavidAnson/markdownlint/pull/1459/commits/02d116629e5417d922046572307c6a6ee0414c6d)!